pex/pip/tool.py

```diff
@@ -358,7 +358,7 @@ def _spawn_pip_isolated(
         with ENV.strip().patch(
             PEX_ROOT=ENV.PEX_ROOT,
             PEX_VERBOSE=str(ENV.PEX_VERBOSE),
-            __PEX_UNVENDORED__="1",
+            __PEX_UNVENDORED__="setuptools",
```

Comment:
The issue before this change was the un-vendoring was global and just above in `pex/pip/download_observer.py`, `pex` was newly exposed to Pip to support patches that use Pex code (the `--exclude` patches leverage this). That `pex` code, including everything under `pex.vendor._vendored` was being exposed which broke `pex.third_party` vendored imports for Python2.7 where the meta-path importer has different precedence than under newer Pythons. Since Pex itself never imports setuptools, exposing just that serves both the Pex patches of Pip (which don;t need pex.third_party.setuptools) and vendored Pip itself (which needs setuptools to build sdists). Ideally Pex would not re-write setuptools imports at all (since it never does a pex.third_party import of the code), but Lambdex _does_ and this change would break Lambdex. A test caught this and serves as a backstop to not break Lambdex users.